shuup/core/api/basket.py

```diff
@@ -137,6 +139,7 @@ class BasketSerializer(serializers.Serializer):
     total_price_of_products = serializers.DecimalField(max_digits=FORMATTED_DECIMAL_FIELD_MAX_DIGITS,
                                                        decimal_places=FORMATTED_DECIMAL_FIELD_DECIMAL_PLACES)
     validation_errors = serializers.SerializerMethodField()
+    extra_data = ExtraDataSerializer()
```

Comment:
maybe here you want to use a [JSONField](http://www.django-rest-framework.org/api-guide/fields/#jsonfield), this way you can put whatever data inside. Otherwise, if some want to customize this, he will need to override the basket viewset, the basket serializer and then the extra data serializer. The serializer is a good one to validate the existence of keys and validate exactly the content set, but as this will be an extra data field, maybe that is not so necessary. Your call :smiley: